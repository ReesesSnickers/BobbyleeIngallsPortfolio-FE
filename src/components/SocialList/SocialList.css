.portfolio-sociallist {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}
