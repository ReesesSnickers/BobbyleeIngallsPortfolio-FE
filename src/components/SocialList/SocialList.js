import React from "react";
import PropTypes from "prop-types";

import IconButton from "../IconButton";

import { SOCIAL_MEDIA } from "../../utility/constants/socialMedia";

import { ReactComponent as FacebookLogo } from "../../assets/Facebook-Logo-Square.svg";
import { ReactComponent as TwitterLogo } from "../../assets/Twitter-Logo-WhiteOnBlue.svg";
import { ReactComponent as LinkedInLogo } from "../../assets/LinkedIn-Icon-Square.svg";

const SocialList = ({ array, customClassName }) => {
  return (
    <nav className={customClassName}>
      {array.map((social, key) => {
        // constants
        const isFacebook = social.type === SOCIAL_MEDIA.FACEBOOK;
        const isTwitter = social.type === SOCIAL_MEDIA.TWITTER;
        const isLinkedIn = social.type === SOCIAL_MEDIA.LINKEDIN;
        let icon = isFacebook ? (
          <FacebookLogo />
        ) : isTwitter ? (
          <TwitterLogo />
        ) : isLinkedIn ? (
          <LinkedInLogo />
        ) : undefined;
        return (
          <IconButton
            key={key}
            icon={icon}
            href={social.url}
            social={social.type}
          />
        );
      })}
      <style jsx>{`
        nav {
          display: flex;
          flex-direction: row;
          justify-content: space-around;
          width: 550px;
        }
      `}</style>
    </nav>
  );
};

export default SocialList;

SocialList.propTypes = {
  array: PropTypes.array,
  customClassName: PropTypes.string,
};
SocialList.defaultProps = {
  customClassName: "",
};
