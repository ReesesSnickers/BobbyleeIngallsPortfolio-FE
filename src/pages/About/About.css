.portfolio-about-wrapper {
  width: 100%;
  height: 100vh;
}

.portfolio-about-divider {
  width: 100%;
  height: auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  margin-top: 70px;
  margin-bottom: 30px;
}
.portfolio-about-dividercircle {
  width: 66px;
  height: 66px;
  border: 5px solid #0e1a3c;
  border-radius: 50%;
  transition: border 0.3s linear;
}
.portfolio-about-dividerline {
  width: 389px;
  height: 5px;
  background-color: #0e1a3c;
  margin: 0px 25px;
  transition: width 0.3s linear;
}

.portfolio-about-upper-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 50px;
  transition: margin 0.3s linear;
}

.portfolio-about-description-wrapper {
  padding: 0px 130px 20px;
  font-size: 30px;
  transition: padding 0.3s linear, margin 0.3s linear;
}

.portfolio-about-avatar {
  padding-top: 10px;
}

.portfolio-about-sociallist {
  width: 550px;
  margin-top: 100px;
}

.portfolio-about-avatar .portfolio-avatar-name {
  transition: font 0.3s linear, margin 0.3s linear;
}
.portfolio-about-avatar .portfolio-avatar-title {
  transition: font 0.3s linear;
}
.portfolio-about-avatar .portfolio-avatar {
  transition: width 0.3s linear, height 0.3s linear;
}
.portfolio-about-avatar {
  transition: padding 0.3s linear;
}

@media screen and (max-width: 1450px) {
  .portfolio-about-dividerline {
    width: 300px;
    transition: width 0.3s linear;
  }
}

@media screen and (max-width: 1270px) {
  .portfolio-about-dividerline {
    width: 225px;
    transition: width 0.3s linear;
  }
}

@media screen and (max-width: 1024px) {
  .portfolio-about-dividerline {
    width: 300px;
    transition: width 0.3s linear;
  }
}

@media screen and (max-width: 930px) {
  .portfolio-about-upper-wrapper {
    margin-top: 80px;
    transition: margin 0.3s linear;
  }
  .portfolio-about-description-wrapper {
    margin-bottom: 140px;
    padding-left: 100px;
    padding-right: 100px;
    transition: padding 0.3s linear, margin 0.3s linear;
  }
  .portfolio-about-sociallist {
    display: none;
  }
  .portfolio-about-avatar .portfolio-avatar-name {
    font-size: 30px;
    margin-top: 20px;
    transition: font 0.3s linear, margin 0.3s linear;
  }
  .portfolio-about-avatar .portfolio-avatar-title {
    font-size: 18px;
    transition: font 0.3s linear;
  }
  .portfolio-about-avatar .portfolio-avatar {
    width: 250px;
    height: 250px;
    transition: width 0.3s linear, height 0.3s linear;
  }
}

@media screen and (max-width: 800px) {
  .portfolio-about-dividerline {
    width: 225px;
    transition: width 0.3s linear;
  }
}

@media screen and (max-width: 700px) {
  .portfolio-about-dividercircle {
    border: 3px solid #0e1a3c;
    transition: border 0.3s linear, height 0.3s linear, width 0.3s linear;
  }
  .portfolio-about-dividerline {
    height: 3px;
    transition: height 0.3s linear, width 0.3s linear;
  }
}

@media screen and (max-width: 630px) {
  .portfolio-about-dividerline {
    width: 175px;
    transition: width 0.3s linear;
  }
}

@media screen and (max-width: 570px) {
  .portfolio-about-description-wrapper {
    padding-left: 50px;
    padding-right: 50px;
    transition: padding 0.3s linear;
  }
}

@media screen and (max-width: 630px) {
  .portfolio-about-dividerline {
    width: 125px;
    transition: width 0.3s linear, height 0.3s linear;
  }
}

@media screen and (max-width: 450px) {
  .portfolio-about-dividerline {
    width: 75px;
    height: 2px;
    transition: width 0.3s linear, height 0.3s linear;
  }
  .portfolio-about-dividercircle {
    width: 53px;
    height: 53px;
    border: 2px solid #0e1a3c;
    border-radius: 50%;
    transition: border 0.3s linear, height 0.3s linear, width 0.3s linear;
  }
}

@media screen and (max-width: 430px) {
  .portfolio-about-description-wrapper {
    padding-left: 30px;
    padding-right: 30px;
    transition: padding 0.3s linear;
  }
}

@media screen and (max-width: 400px) {
  .portfolio-about-description-wrapper {
    font-size: 20px;
    transition: font 0.3s linear;
  }
  .portfolio-about-avatar .portfolio-avatar-name {
    font-size: 25px;
    transition: font 0.3s linear;
  }
  .portfolio-about-avatar .portfolio-avatar-title {
    font-size: 18px;
    transition: font 0.3s linear;
  }
  .portfolio-about-avatar .portfolio-avatar {
    width: 200px;
    height: 200px;
    transition: width 0.3s linear, height 0.3s linear;
  }
}
