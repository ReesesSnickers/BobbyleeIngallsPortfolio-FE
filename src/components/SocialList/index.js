import "./SocialList.css";
export { default } from "./SocialList";
