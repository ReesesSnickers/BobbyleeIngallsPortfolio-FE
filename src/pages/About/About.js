import React from "react";

// config
import socialConfig from "../../configs/socialConfig";
import {
  paragraph1,
  paragraph2,
  paragraph3
} from "../../configs/aboutMeConfig";

// components
import Avatar from "../../components/Avatar";
import SocialList from "../../components/SocialList";

const About = ({ photo }) => {
  return (
    <section className="portfolio-about-wrapper">
      <section className="portfolio-about-upper-wrapper">
        <div className="portfolio-about-avatar">
          <Avatar
            src={photo}
            alt="test"
            name="Bobbylee Ingalls"
            title="Software Engineer"
            border
          />
        </div>
        <div className="portfolio-about-sociallist">
          <SocialList array={socialConfig} />
        </div>
        <div className="portfolio-about-divider">
          <div className="portfolio-about-dividerline" />
          <div className="portfolio-about-dividercircle" />
          <div className="portfolio-about-dividerline" />
        </div>
      </section>
      <section className="portfolio-about-description-wrapper">
        <p>{paragraph1}</p>
        <p>{paragraph2}</p>
        <p>{paragraph3}</p>
      </section>
    </section>
  );
};

export default About;
