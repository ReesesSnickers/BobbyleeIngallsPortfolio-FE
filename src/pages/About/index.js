import "./About.css";
export { default } from "./About";
