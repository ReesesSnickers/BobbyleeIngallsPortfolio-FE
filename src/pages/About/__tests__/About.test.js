import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import About from "../About.js";
import {
  paragraph1,
  paragraph2,
  paragraph3
} from "../../../configs/aboutMeConfig";
import SocialList from "../../../components/SocialList";

let props;
let mountedComponent;

describe("About", () => {
  const testComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<About {...props} />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    props = {
      photo: "MockPhotoString"
    };
    mountedComponent = undefined;
  });

  // initial render
  it("Renders the About component", () => {
    let section = testComponent().find("section.portfolio-about-wrapper");
    expect(section.length).toBe(1);
  });

  // snapshot
  it("should match snapshot", () => {
    const tree = renderer.create(<About {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // photo
  it(`passes the photo prop to Avatar's src prop`, () => {
    const Avatar = testComponent().find("Avatar");
    expect(Avatar.props().src).toBe("MockPhotoString");
  });
});
