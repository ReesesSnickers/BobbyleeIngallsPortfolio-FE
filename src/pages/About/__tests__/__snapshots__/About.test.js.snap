// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match snapshot 1`] = `
<section
  className="portfolio-about-wrapper"
>
  <section
    className="portfolio-about-upper-wrapper"
  >
    <div
      className="portfolio-about-avatar"
    >
      <section
        className="portfolio-avatar-wrapper"
      >
        <section
          className="portfolio-avatar portfolio-avatar-border"
        >
          <img
            alt="test"
            className="portfolio-avatar-image"
            src="MockPhotoString"
          />
        </section>
        <section
          className="portfolio-avatar-user"
        >
          <h1
            className="portfolio-avatar-name"
          >
            Bobbylee Ingalls
          </h1>
          <h2
            className="portfolio-avatar-title"
          >
            Software Engineer
          </h2>
        </section>
      </section>
    </div>
    <div
      className="portfolio-about-sociallist"
    >
      <section
        className="portfolio-sociallist"
      >
        <a
          className="portfolio-IconButton"
          href="https://www.linkedin.com/in/bobbyleeingalls/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg>
            LinkedIn-Icon-Square.svg
          </svg>
        </a>
        <a
          className="portfolio-IconButton"
          href="https://twitter.com/BobbyleeIngalls"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg>
            Twitter-Logo-WhiteOnBlue.svg
          </svg>
        </a>
        <a
          className="portfolio-IconButton"
          href="https://www.facebook.com/Bobbylee-Ingalls-102087861139492/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg>
            Facebook-Logo-Square.svg
          </svg>
        </a>
      </section>
    </div>
    <div
      className="portfolio-about-divider"
    >
      <div
        className="portfolio-about-dividerline"
      />
      <div
        className="portfolio-about-dividercircle"
      />
      <div
        className="portfolio-about-dividerline"
      />
    </div>
  </section>
  <section
    className="portfolio-about-description-wrapper"
  >
    <p>
      First I must thank you for taking the time to view my portfolio. My name is Bobbylee Ingalls. I am a self-taught software developer who’s primary expertise is in front-end ReactJs development but I have some understanding in NodeJs backend development. I say I was self-taught but I had some great mentor’s that helped guide my learning and mold me to the developer I am today. So I need to give a big shout out to Van Tabbert and Roger Hernandez for being awesome mentors! So more about me. I Love outdoor activities such as camping, hiking and kayaking, although I find it hard sometimes to get out to do them. Most my time is spent developing and spending time with my family little family consisting of my wife, two daughters, two dogs, a cat and myself.
    </p>
    <p>
      Another notable fact about me was I did not start in a technology career field. I actually served six years active duty with the United States Air Force as a KC-135 Crew Chief.  As my enlistment came closer to an end I found myself struggling to know what I wanted to do next and limiting my searches to Logistic career fields as I felt that’s all I knew to do. Luckily I had someone tell me I was smart and could learn anything I put my mind to and helped me get a job with Wall-Mart as a Home Office Client Helpdesk technician. It was a foot in the door that led me to learning powershelgl to help script out typical tasks to increase productivity of our technicians and decrease the time it took for the customer to go from issue to having the issue be resolved. This led me to a sub group in the team who were building a self service platform for the team and the wished to use my powershelgl knowledge to help automate some tasks. It didn’t take long before I was pushed in the direction to learn React and start a new path to learning software development. Not to long after that the sub team I was working with was merged into one of Wal-Marts dev teams building applications to help the productivity of Home Office associates. And I have been doing that ever since.
    </p>
    <p>
      Now that you know my story. I hope that inspires you. You don’t HAVE to obtain a degree to be a developer. It just takes persistence to learn what you don’t know or understand. Always stride for excellence and look for good mentor’s. You can do whatever you set your mind too.
    </p>
  </section>
</section>
`;
